hazelcast/src/main/java/com/hazelcast/client/impl/clientside/ClientDiscoveryServiceBuilder.java

```diff
@@ -162,13 +159,19 @@ private AddressProvider createAddressProvider(ClientConfig clientConfig, Discove
     }
 
     private boolean discoverySpiEnabled(ClientNetworkConfig networkConfig) {
-        return (networkConfig.getDiscoveryConfig() != null && networkConfig.getDiscoveryConfig().isEnabled())
-                || Boolean.parseBoolean(properties.getString(DISCOVERY_SPI_ENABLED));
+        DiscoveryConfig discoveryConfig = networkConfig.getDiscoveryConfig();
+        if (discoveryConfig.getDiscoveryServiceProvider() != null) {
+            return true;
+        }
+        Collection<DiscoveryStrategyConfig> discoveryStrategyConfigs = discoveryConfig.getDiscoveryStrategyConfigs();
+        if (discoveryStrategyConfigs.stream().anyMatch(DiscoveryStrategyConfig::isEnabled)) {
+            return true;
+        }
+        return networkConfig.getDiscoveryConfig() != null && networkConfig.getDiscoveryConfig().isEnabled();
```

Comment:
I think this condition does not make sense here, `networkConfig.getDiscoveryConfig() != null`, you already know it's not `null`, otherwise you'd get `NullPointerException` in the line `163`.

Reply:
It looks like, it missed my attention when I was merging two different methods. Updating accordingly.